### PR TITLE
chore(ci): extract coverage upload into separate job (VIV-1136)

### PR DESCRIPTION
**Pull request by RichardHelm**

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -46,11 +46,18 @@ jobs:
...
       os: ubuntu-latest
       report_coverage: true
 
+  upload-coverage:
+    if: github.event.pull_request.draft == false
+    needs: unit-tests
+    uses: vonage/vivid-3/.github/workflows/_upload-coverage.yml@039a315116ee99695d3f8f2dcdd245282aa5f02a
+    with:
+      os: ubuntu-latest
+
```
Comment (YonatanKra) [inline .github/workflows/pull-request.yml:60]: We can move to relative paths so we wouldn't need to state the branch/commit hash.
See here:
https://github.com/Vonage/vivid-3/blob/main/.github/workflows/release-please.yml#L90

**Review by YonatanKra — CHANGES_REQUESTED** [review]
Great job!
Thanks!
Just a small change :)

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (RichardHelm) [reply .github/workflows/pull-request.yml]: Ah, that's much better
Updated 👍

**Review by YonatanKra — APPROVED** [review]
Great stuff! Thanks!

**Pull request merged**